### ISSUE TITLE
custom `current_user` with arity > 0 raises exception

### ISSUE THREAD

**Issue by mltsy** [airbrake/airbrake#618]
* Airbrake version: 5.5.0
* Ruby version: 2.1.10
* Framework name & version: Rails 4.2

#### Airbrake config

Airbrake.configure do |config|
  config.project_key = 'xxx'
  config.project_id = 'xxx'
  config.host = 'errbit.iexposure.com'
  config.environment = Rails.env == 'production' ? Rails.root.to_s.gsub(/\/home\/rails(?:-staging)?\/(.+?)\..+\z/mi, '\1') : Rails.env
end

#### Description

If you have a `current_user` method in your controller, Airbrake will attempt to call it with 0 arguments, regardless of how many arguments are required.

This is, of course, an edge case, but there are reasons it may be valid (in our case, we have an old app that uses two completely separate sets of user models for admin and client users, so you have to tell current_user which type of user you're looking for)

The simple fix would be to add an arity check before calling current_user:

```ruby
# Note: a method with only optional arguments will have arity == -1
if [-1,0].include(controller.method(:current_user).arity)
  ... # call it, etc.
end
```

**Comment by kyrylo** [comment]
How exactly can Airbrake know what to pass to `current_user`?
My best suggestion is to refactor that method, so it can work without arguments (and default to some type of users).

**Comment by mltsy** [comment]
Sorry, hah, let me clarify :)  Basically, I'm just saying I should be able to add an arbitrary current_user method to my controller without crashing the app!  My suggestion is just to skip calling the method if it doesn't work with 0 arguments (as if the method didn't exist - technically, a method with the correct signature doesn't exist).

**Comment by kyrylo** [comment]
Could you provide a backtrace of a crash? Feel free to send it to kyrylo@airbrake.io if it's private.

**Comment by mltsy** [comment]
Well, I've fixed the issue in my app (by working around it similarly to what you suggested), but I do remember that it was as a result of an exception that was trying to report itself in my app, and the top of the stack trace for the subsequent exception was: https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rack/user.rb#L22

If you need a stack trace for some reason I could probably reproduce one... is there some other information you'd like from it?  I'll submit a PR for my suggested fix.

**Comment by kyrylo** [comment]
Backtrace can be omitted if we have a failing test. I just want to verify it's an issue. We can either do nothing (add an entry to the README with a warning about `current_method`) or apply your idea. I am currently on the fence.

**Comment by kyrylo** [comment]
@mltsy I just released v5.6.1. Enjoy!
